gramineproject/gramine
# [Pal/Linux,Linux-SGX] Make sure `g_debug_map` is complete

## Description of the changes 

After this change, `g_debug_map` contains all loaded executables, even the ones that were loaded by the system (such as main Gramine executable or protobuf).

This is not necessary for GDB integration, but will help add features such as crash location reporting (and it already helped simplify the SGX profiling system).

(This is in preparation to add source line reporting to ASan. See discussion in #32 for context: we have a raw pointer, will probably run `addr2line` in a subprocess, this is necessary for determining the right file and offset).

## How to test this PR? 

* I checked using GDB that `g_debug_map` contains all the debug maps (including `libpal.so`, `pal-sgx`, etc.)  in both Linux and Linux-SGX modes
* SGX profiling should still work, the change influences reporting _outer_ maps (so I tested that with `sgx.profile.mode = "ocall_outer" the report still contains symbols from URTS)


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/gramine/111)
